### ISSUE TITLE
load.ivy through HTTP/HTTPS proxy for Spark setup

### ISSUE THREAD

**Issue by rvernica** [alexarchambault/jupyter-scala#21]
I am following the Spark setup instructions listed in [Spark.ipynb](https://github.com/alexarchambault/jupyter-scala/blob/topic/spark-example/examples/libraries/Spark.ipynb). How can I configure `load.ivy` to go through an HTTP/HTTPS proxy for donwloading?

Thanks!

**Comment by rvernica** [comment]
If I set the HTTP/HTTPS proxy in `SBT_OPTS` before starting Ammonite, the the `load.ivy` call works from Ammonite:

```
SBT_OPTS="-Dhttp.proxyHost=... -Dhttp.proxyPort=... -Dhttps.proxyHost=... -Dhttps.proxyPort=..." ./amm
Loading Ammonite Repl...
@ load.ivy("com.github.alexarchambault" % "ammonite-spark_1.3_2.10.5" % "0.3.1-S  NAPSHOT")

@ 
```

**Comment by alexarchambault** [comment]
You can try editing `~/.ipython/kernels/scala211/kernel.json`, then, and add the options there:
```json
{
  "language" : "scala211",
  "display_name" : "Scala 2.11",
  "argv" : [
    "/path/to/jupyter-scala_2.11.6-0.2.0-SNAPSHOT/bin/jupyter-scala",
    "--quiet",
    "--connection-file",
    "{connection_file}",
    "-Dhttp.proxyHost=...",
    "-Dhttp.proxyPort=...",
    "-Dhttps.proxyHost=...",
    "-Dhttps.proxyPort=..."
  ]
}
```

**Comment by alexarchambault** [comment]
Sorry, wrong hint! The `jupyter-scala` script doesn't take java options this way. You can try to edit this script directly (its path is in `kernel.json` above), and the options at the end, before the current java options (like `-D...`).

**Comment by rvernica** [comment]
Thanks! It worked.

**Comment by cequencer** [comment]
I noticed this solution worked 3 years ago, but now the jupyter-scala is using couriser to download all the runtime dependencies and artifacts.  I have tried the proposed method here, but it no longer works with couriser.  I have provided more attempts and clues in issue #189.  It will be great if @alexarchambault can revisit this either issue with the new codebase using couriser.  Many thanks and sorry for trolling an closed issue from the past.
